### ISSUE TITLE
"ValueError: list.remove(x): x not in list" logged during runtime

### ISSUE THREAD

**Issue by olii** [NoneGG/aredis#140]
## Checklist

- Python version 3.8
- hiredis
- asyncio event loop
- Does issue exists against the `master` branch of aredis? Tested with latest release 1.1.7

## Steps to reproduce
1. Connect to redis sentinel and set the `max_idle_time` as connection kwarg.
2. Get the master from sentinel.
3. Use master instance.

## Expected behavior
No exception is raised.

## Actual behavior
```
ERROR:asyncio:Task exception was never retrieved
future: <Task finished coro=<ConnectionPool.disconnect_on_idle_time_exceeded() done, defined at /usr/local/lib/python3.7/dist-packages/aredis/pool.py:183> exception=ValueError('list.remove(x): x not in list')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/aredis/pool.py", line 188, in disconnect_on_idle_time_exceeded
    self._available_connections.remove(connection)
ValueError: list.remove(x): x not in list
```

**Comment by NoneGG** [comment]
@olii Thank you for bug report, it seems like connection is disconnected before, i will add check on `available_connections` to avoid this error being raised again.

**Comment by NoneGG** [comment]
Feel free to reopen this issue if the problem is not solved~

**Comment by cryeo** [comment]
Same phenomenon was observed in cluster connection pool.

```
ERROR Task exception was never retrieved
future: <Task finished name='Task-2456602363' coro=<ClusterConnectionPool.disconnect_on_idle_time_exceeded() done, defined at /usr/local/pyenv/versions/py382/lib/python3.8/site-packages/aredis/pool.py:326> exception=ValueError('list.remove(x): x not in list')>
Traceback (most recent call last):
 File "/usr/local/pyenv/versions/py382/lib/python3.8/site-packages/aredis/pool.py", line 332, in disconnect_on_idle_time_exceeded
   self._available_connections[node['name']].remove(connection)
ValueError: list.remove(x): x not in list
```

I think that it can be fixed in the same way as #144.
